### PR TITLE
[Frontendv1] Dynamic width for web container.

### PR DESCRIPTION
**Pull request by Alabhya268**
fixed issue #3254 .

Web container and challengelist grid is now dynamic and footer and other pages are also utilizing full screen width .

For screenshots of before please refer to issue.

AFTER -
after3
after4
after2
after1

### CODE REVIEW

**Comment by Alabhya268** [comment]
@RishabhJain2018 @Ram81 , Please review this PR.

**Comment by Alabhya268** [comment]
@RishabhJain2018 @Ram81 , Please review this PR.

File: frontend/src/css/modules/base.scss [hunk 1]
```diff
@@ -1032,4 +1038,4 @@ h4.challenge-step-title {
         z-index: 10;
         position: relative;
     }
-}
+}
```
Comment (Ram81) [inline frontend/src/css/modules/base.scss:1041]: @Alabhya268 can you add a newline

File: frontend/src/css/modules/base.scss [hunk 2]
```diff
@@ -1006,16 +1006,22 @@ h4.challenge-step-title {
         background: white;
     }
 
+    :root {
+        .col.m3 {
```
Comment (Ram81) [inline frontend/src/css/modules/base.scss:1010]: @Alabhya268 why is this done just for `col m3` we will need it for other column sizes too, no?

**Review by Ram81 — CHANGES_REQUESTED** [review]

File: frontend/src/css/modules/base.scss (hunk 1 above)
Comment (Alabhya268) [reply frontend/src/css/modules/base.scss:1041]: ok

File: frontend/src/css/modules/base.scss (hunk 2 above)
Comment (Alabhya268) [reply frontend/src/css/modules/base.scss:1010]: @Ram81 , 'col m3'  is used for challenge cards, Therefore it is unecessary to change sizes for other columns.

**Comment by Alabhya268** [comment]
@Ram81 Please review this PR again.

**Comment by Alabhya268** [comment]
@Ram81 , Ping for this PR.

File: frontend/src/css/modules/base.scss [hunk 3]
```diff
@@ -1033,3 +1039,4 @@ h4.challenge-step-title {
         position: relative;
     }
 }
+
```
Comment (Ram81) [inline frontend/src/css/modules/base.scss:1042]: @Alabhya268 can you remove extra newline. Rest of it LGTM

**Review by Ram81 — CHANGES_REQUESTED** [review]

File: frontend/src/css/modules/base.scss (hunk 3 above)
Comment (Alabhya268) [reply frontend/src/css/modules/base.scss:1042]: ok

Comment (Alabhya268) [reply frontend/src/css/modules/base.scss]: @Ram81 ,I  have removed this line now.Let me know if anything else needs to be changed.

**Comment by Alabhya268** [comment]
@Ram81 , Ping for PR.

**Comment by Ram81** [comment]
@Alabhya268 let's not do this change of changing to dynamic width container for frontend v1 if it behaves same as #3253

**Comment by Alabhya268** [comment]
> @Alabhya268 let's not do this change of changing to dynamic width container for frontend v1 if it behaves same as #3253

@Ram81 , There is no issue in this PR like dynamic width container for frontend v1.
These are the screenshots taken in 1080p screen width just now.

**Comment by Alabhya268** [comment]
@Ram81 , Ping for Review.

**Comment by Ram81** [comment]
@Alabhya268 can you try testing by changing the browser zoom property and confirm if everything works fine.

**Comment by Alabhya268** [comment]
@Ram81 , Could you also help me with few queries regarding GSOC. As student I am really interested in participating.
Also for some reason mailing list is not working .

**Comment by Alabhya268** [comment]
@Ram81 , Its working completely please look at this video .
https://user-images.githubusercontent.com/57143358/110642868-77a32900-81d9-11eb-9394-d1692a75ee57.mp4

**Comment by Ram81** [comment]
@Alabhya268 Even for frontend v1 the cards start getting bigger in width 
Screen Shot 2021-03-11 at 12 39 04 AM

Screen Shot 2021-03-11 at 12 38 56 AM

**Comment by Ram81** [comment]
>@Ram81 , Could you also help me with few queries regarding GSOC. As student I am really interested in participating.

Please post your queries on the gitter channel. We are happy to answer your queries.

>Also for some reason mailing list is not working .

Yes, we are looking into the issue

**Comment by Alabhya268** [comment]
> fixed issue #3254 .
> 
> Web container and challengelist grid is now dynamic and footer and other pages are also utilizing full screen width .
> 
> For screenshots of before please refer to issue.
> 
> AFTER -
> after3
> after4
> after2
> after1

@Ram81 , I finally found the problem , As you can see in my pc images were not showing prevoiusly which was causing expanding of cards . I have fixed this issue.

Could you please review this PR again.

**Comment by RishabhJain2018** [comment]
Hi @Alabhya268,
Thank you for the effort but it is a design choice. We actually don't want to make the challenge cards scale dynamically. We would like to keep it centered only.

Please feel free to pick up another issue. Also, please be on the lookout as @Ram81 will open more issues in the next few days.

**Pull request closed without merging**